File: python/Sliding_window/lc3_longest_unique_substring.py
"""
3.无重复字符的最长子串
给定一个字符串 s ，请你找出其中不含有重复字符的 最长 子串 的长度。

Date: 2025-11-4
"""
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        dic = {}
        res = 0
        left = 0
        for i in range(len(s)):
            if s[i] in dic:
                left = max(left, dic[s[i]] + 1)
            dic[s[i]] = i
            res = max(res, i - left + 1)
        return res
if __name__ == "__main__":
    sol = Solution()
    s1 = "abcabcbb"
    print(sol.lengthOfLongestSubstring(s1))

File: c_plus_plus/Sliding_window/lc3_longest_unique_substring.cpp
/*
3.无重复字符最长子串
给定一个字符串 s ，请你找出其中不含有重复字符的 最长 子串 的长度。

date:2026-04-09
*/
#include<iostream>
#include<stdio.h>
#include<vector>
#include<string>
#include<unordered_map>
using namespace std;

int lengthOfLongestSubstring(string s)
{
    int n = s.length();
    unordered_map<int> mp;
    int left = 0,right = 0;
    while(left < right){
        ms[s[right]]++;
    }
    return 0;
}
int main()
{
    string s = "abcabcbb";
    cout << lengthOfLongestSubstring(s) << endl;
    
    return 0;
    
}